core/src/main/java/cucumber/runtime/formatter/UnusedStepsSummaryPrinter.java

```diff
@@ -0,0 +1,76 @@
+package cucumber.runtime.formatter;
+
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+import cucumber.api.Result;
+import cucumber.api.SummaryPrinter;
+import cucumber.api.event.*;
+import cucumber.api.formatter.ColorAware;
+import cucumber.api.formatter.NiceAppendable;
+
+public class UnusedStepsSummaryPrinter implements ColorAware, EventListener, SummaryPrinter {
+
+	private EventHandler<StepDefinedEvent> stepDefinedHandler = new EventHandler<StepDefinedEvent>() {
+		@Override
+		public void receive(StepDefinedEvent event) {
+			unusedSteps.put(event.stepDefinition.getLocation(false), event.stepDefinition.getPattern());
+		}
+	};
+	private EventHandler<TestStepFinished> testStepFinishedHandler = new EventHandler<TestStepFinished>() {
+		@Override
+		public void receive(TestStepFinished event) {
+			String codeLocation = event.testStep.getCodeLocation();
+			if (codeLocation != null) {
+				unusedSteps.remove(codeLocation);
+			}
+		}
+	};
+	private EventHandler<TestRunFinished> testRunFinishedhandler = new EventHandler<TestRunFinished>() {
+		@Override
+		public void receive(TestRunFinished event) {
+			if (unusedSteps.isEmpty()) {
+				return;
+			}
+
+			Format format = formats.get(Result.Type.UNUSED.lowerCaseName());
+			out.println(format.text(unusedSteps.size() + " Unused steps:"));
+
+			// Output results when done
+			for (Entry<String, String> entry : unusedSteps.entrySet()) {
+				String location = entry.getKey();
+				String pattern = entry.getValue();
+				out.println(format.text(location + " # " + pattern));
```

Comment:
```suggestion
				out.println(format.text(location) + " # " + pattern);
```

Only the location is in yellow.